pyomo/core/plugins/transform/strip_bounds.py

```diff
@@ -0,0 +1,68 @@
+"""Transformation to strip variable bounds from a model."""
+import textwrap
+
+from pyomo.core.base.var import Var
+from pyomo.core.plugins.transform.hierarchy import IsomorphicTransformation
+from pyomo.util.plugin import alias
+from pyomo.core.kernel.component_map import ComponentMap
+from six import iteritems
+from pyomo.core.kernel.set_types import Reals
+
+__author__ = "Qi Chen <https://github.com/qtothec>"
+
+
+class VariableBoundStripper(IsomorphicTransformation):
+    """Strips bounds from variables."""
+
+    alias('core.strip_var_bounds',
+          doc=textwrap.fill(textwrap.dedent(__doc__.strip())))
+
+    def __init__(self):
+        """Initialize the transformation."""
+        super(VariableBoundStripper, self).__init__()
+
+    def _apply_to(self, instance, strip_domains=True, tmp=False):
+        """Apply the transformation.
+
+        Args:
+            instance (Block): the block on which to strip variable bounds
+            strip_domains (bool, optional): strip the domain for discrete
+                variables as well
+            tmp (bool, optional): Whether the bound stripping will be
+                temporary. If so, store information for reversion.
+
+        Returns:
+            None
+
+        """
+        if tmp and not hasattr(instance, '_tmp_var_bound_strip_lb'):
+            instance._tmp_var_bound_strip_lb = ComponentMap()
+        if tmp and not hasattr(instance, '_tmp_var_bound_strip_ub'):
+            instance._tmp_var_bound_strip_ub = ComponentMap()
+        if tmp and not hasattr(instance, '_tmp_var_bound_strip_domain'):
+            instance._tmp_var_bound_strip_domain = ComponentMap()
+        for var in instance.component_data_objects(ctype=Var):
+            if strip_domains and not var.domain == Reals:
+                if tmp:
+                    instance._tmp_var_bound_strip_domain[var] = var.domain
+                var.domain = Reals
+            if var.has_lb():
+                if tmp:
+                    instance._tmp_var_bound_strip_lb[var] = var.lb
+                var.setlb(None)
+            if var.has_ub():
+                if tmp:
+                    instance._tmp_var_bound_strip_ub[var] = var.ub
+                var.setub(None)
+
+    def revert(self, instance):
+        """Revert variables fixed by the transformation."""
+        for var, lb in iteritems(instance._tmp_var_bound_strip_lb):
```

Comment:
Will you ever run into a case where a variable has been completely removed from the instance between applying this transformation and reverting it? Would it be safer to iterate over all the Var Data objects again and look-up that object in the component maps?

Reply:
Completely removing a variable from an instance sounds like asking for trouble to me

Reply:
Alas, this is why we need the design discussion for transformations - should do it soon. I think it is reasonable to add a condition (assumption) that the model structure has not changed between these calls. If a user wants that behavior, they can write a specialized "revert" using the data.

Reply:
I'll switch to doing the lookups, since I don't feel strongly enough about this either way right now.